### PR TITLE
Parquet: Avoid modifying existing conf of HadoopOutputFile rather create new one

### PR DESCRIPTION
**Pull request by singhpk234**
we faced concurrent modification issue in ORC when trying to modify existing hadoop conf from HadoopFileIO in ORC write builder.  ref : [Issue#4383](https://github.com/apache/iceberg/issues/4383), [PR#4384](https://github.com/apache/iceberg/pull/4384)

Looks like this code snippet was present in Parquet#write as well, based on discussion https://github.com/apache/iceberg/pull/3810#issuecomment-1061334692

As per my understanding, we should do the same changes as we did in the ORC, here in parquet. The issue we had in ORC (given that we expect our PR fixed it), could also happen in Parquet#write under same circumstances, added a test for the same

----
cc @openinx, @hililiwei , @rdblue

### CODE REVIEW

**Comment by hililiwei** [comment]
Thanks for the fix.

File: parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java [hunk 1]
```diff
@@ -258,15 +264,9 @@ private WriteBuilder createContextFunc(Function<Map<String, String>, Context> ne
...
-        Configuration conf;
-        if (file instanceof HadoopOutputFile) {
-          conf = ((HadoopOutputFile) file).getConf();
-        } else {
-          conf = new Configuration();
-        }
 
         for (Map.Entry<String, String> entry : config.entrySet()) {
-          conf.set(entry.getKey(), entry.getValue());
+          this.conf.set(entry.getKey(), entry.getValue());
```
Comment (jackye1995) [inline parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java:269]: this looks like unnecessary change?

File: spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java [hunk 2]
```diff
@@ -85,6 +85,15 @@ public SparkRowLevelOperationsTestBase(String catalogName, String implementation
             true,
             WRITE_DISTRIBUTION_MODE_NONE
         },
+        { "testhive", SparkCatalog.class.getName(),
+          ImmutableMap.of(
```
Comment (jackye1995) [inline spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/SparkRowLevelOperationsTestBase.java:89]: indentation seems off

**Review by rajarshisarkar — APPROVED** [review]

**Comment by singhpk234** [comment]
@openinx can you please take a pass when you have some time

**Review by openinx — APPROVED** [review]

**Comment by openinx** [comment]
Got this merged, thanks @singhpk234 for the contribution !
